app/models/hostgroup.rb

```diff
@@ -232,4 +235,8 @@ def validate_subnet_types
     errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if self.subnet.present? && self.subnet.type != 'Subnet::Ipv4'
     errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if self.subnet6.present? && self.subnet6.type != 'Subnet::Ipv6'
   end
+
+  def suggest_default_pxe_loader
+    self.pxe_loader = self.try(:operatingsystem).try(:preferred_loader) unless self.pxe_loader.present?
```

Comment:
`pxe_loader ||= self.try(:operatingsystem).try(:preferred_loader)` - quite a violation of the law of demeter though